boa_engine/src/builtins/array/mod.rs

```diff
@@ -1628,35 +1603,20 @@ impl Array {
         // 2. Let len be ? LengthOfArrayLike(O).
         let len = o.length_of_array_like(context)?;
 
-        // 3. If IsCallable(predicate) is false, throw a TypeError exception.
         let predicate = args.get_or_undefined(0).as_callable().ok_or_else(|| {
             JsNativeError::typ().with_message("Array.prototype.findLast: predicate is not callable")
         })?;
-
         let this_arg = args.get_or_undefined(1);
 
-        // 4. Let k be len - 1. (implementation differs slightly from spec because k is unsigned)
-        // 5. Repeat, while k >= 0, (implementation differs slightly from spec because k is unsigned)
-        for k in (0..len).rev() {
-            // a. Let Pk be ! ToString(𝔽(k)).
-            // b. Let kValue be ? Get(O, Pk).
-            let k_value = o.get(k, context)?;
-            // c. Let testResult be ! ToBoolean(? Call(predicate, thisArg, « kValue, 𝔽(k), O »)).
-            let test_result = predicate
-                .call(
-                    this_arg,
-                    &[k_value.clone(), k.into(), this.clone()],
-                    context,
-                )?
-                .to_boolean();
-            // d. If testResult is true, return kValue.
-            if test_result {
-                return Ok(k_value);
-            }
-            // e. Set k to k - 1.
+        // 3. Let findRec be ? FindViaPredicate(O, len, descending, predicate, thisArg).
+        let find_rec =
+            find_via_predicate(&o, len, Direction::Descending, predicate, this_arg, context);
+
+        // 4. Return findRec.[[Value]].
+        match find_rec {
+            Ok((_, value)) => Ok(value),
+            Err(err) => Err(err),
```

Comment:
```suggestion
        let (_, value) = find_via_predicate(&o, len, Direction::Descending, predicate, this_arg, context)?;

        // 4. Return findRec.[[Value]].
        Ok(value)
```

We can use `?` error propagation and `let` pattern matching to remove `match` here, same for other call sites

Reply:
Thanks for the review! Applied in fd1af53 looks much cleaner now ;)